examples/model_selection/plot_cost_sensitive_learning.py

```diff
@@ -0,0 +1,831 @@
+# feature, we are only required to have a column in `data` that contains the amount of
+# each transaction. To use this information as metadata, we need to have an external
+# variable that we can pass to the scorer or the model that internally routes this
+# metadata to the scorer. So let's create this variable.
+amount = credit_card.frame["Amount"].to_numpy()
+
+# %%
+# We first start to train a dummy classifier to have some baseline results.
+from sklearn.model_selection import train_test_split
+
+data_train, data_test, target_train, target_test, amount_train, amount_test = (
+    train_test_split(
+        data, target, amount, stratify=target, test_size=0.5, random_state=42
+    )
+)
+
+# %%
+from sklearn.dummy import DummyClassifier
+
+easy_going_classifier = DummyClassifier(strategy="constant", constant=0)
+easy_going_classifier.fit(data_train, target_train)
+benefit_cost = business_scorer(
+    easy_going_classifier, data_test, target_test, amount=amount_test
+)
+print(f"Benefit/cost of our easy-going classifier: ${benefit_cost:,.2f}")
+
+# %%
+# A classifier that predict all transactions as legitimate would create a profit of
+# around $220,000. We make the same evaluation for a classifier that predicts all
+# transactions as fraudulent.
+intolerant_classifier = DummyClassifier(strategy="constant", constant=1)
+intolerant_classifier.fit(data_train, target_train)
+benefit_cost = business_scorer(
+    intolerant_classifier, data_test, target_test, amount=amount_test
+)
+print(f"Benefit/cost of our intolerant classifier: ${benefit_cost:,.2f}")
+
+# %%
+# Such a classifier create a loss of around $670,000. A predictive model should allow
+# us to make a profit larger than $220,000. It is interesting to compare this business
+# metric with another "standard" statistical metric such as the balanced accuracy.
+from sklearn.metrics import get_scorer
+
+balanced_accuracy_scorer = get_scorer("balanced_accuracy")
+print(
+    "Balanced accuracy of our easy-going classifier: "
+    f"{balanced_accuracy_scorer(easy_going_classifier, data_test, target_test):.3f}"
+)
+print(
+    "Balanced accuracy of our intolerant classifier: "
+    f"{balanced_accuracy_scorer(intolerant_classifier, data_test, target_test):.3f}"
+)
+
+# %%
+# This is not a surprise that the balanced accuracy is at 0.5 for both classifiers.
+# However, we need to be careful in the rest of the evaluation: we potentially can
+# obtain a model with a decent balanced accuracy that does not make any profit.
+# In this case, the model would be harmful for our business.
+#
+# Let's now create a predictive model using a logistic regression without tuning the
+# decision threshold.
+from sklearn.linear_model import LogisticRegressionCV
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+
+model = make_pipeline(
+    StandardScaler(),
+    LogisticRegressionCV(random_state=42, Cs=np.logspace(-6, 6, 13), scoring="roc_auc"),
```

Comment:
can we please not do StandardScaler before a CV object? It's leaking data. Rather have this inside a GridSeachCV.

Reply:
I don't think it's possible to get a significant data leak with standard scaler.

Personally I would prefer keeping `LogisticRegressionCV`. It makes the example more concise and run faster thanks to the inner warm starts.

Reply:
Note that if we implement what you suggest in https://github.com/scikit-learn/scikit-learn/pull/26120/files#r1586058340 we could only tune `search_cv.best_estimator_` instead of the redoing the grid search for each CV split of the tuner. In this case using a `GridSearchCV` should be fast enough for this example because we would run it only once (and it provides the `.best_estimator_` attribute which makes this approach quite convenient).

Reply:
Real data has real outliers, and unless one properly investigates the data, using this pattern as is in this example should really be avoided. I already showed how outliers break things here: https://github.com/scikit-learn/scikit-learn/issues/26359#issuecomment-1571838168

Reply:
Here, I see that we use "roc_auc", we should switch to "neg_log_loss" (this improve the results) and it is inline with recent insights that we got. Not scaling will have huge impact on the solver.

So if outlier is actually an issue, we should be using `RobustScaler` (that also provide strong results here). And on the original post, it will provide what you expect:

```python
In [2]: from sklearn.datasets import make_regression
   ...: from sklearn.linear_model import Lasso, LassoCV
   ...: from sklearn.pipeline import make_pipeline
   ...: from sklearn.preprocessing import RobustScaler
   ...: from sklearn.model_selection import GridSearchCV
   ...: 
   ...: X, y = make_regression(n_samples=200, n_features=500, n_informative=5, random_state=0)
   ...: X[:2, ] = X[:2, ] +20
   ...: 
   ...: est = make_pipeline(RobustScaler(quantile_range=(5, 95)), LassoCV(cv=5)).fit(X, y)
   ...: 
   ...: est_gs = GridSearchCV(
   ...:     make_pipeline(RobustScaler(quantile_range=(5, 95)), Lasso(random_state=0)),
   ...:     param_grid={"lasso__alpha": est[-1].alphas_},
   ...:     cv=5,
   ...: ).fit(X, y)
   ...: print(est[-1].alpha_)
   ...: print(est_gs.best_estimator_[-1].alpha)
```

```
30.119681403085156
30.119681403085156
```

Reply:
In short, the leak here will have an impact only if we have a sort of distribution shift between folds (or train/test sets). Having a robust estimator for the scaling should always cope with this issue if the shift is due to outlier.

Reply:
Eventually, we can an explicit note about this specific pipeline and choice of scaler.

Reply:
In this case yes, but using RobustScaler is not always a better solution, if it was, we would ditch StandardScaler and always recommend RobustScaler, but we don't, cause in many cases those outliers are important somehow. I really don't see why we should include something in our examples that could have data leakage issues. We should always recommend the safer way. If people want to optimize for speed and really know what they're doing and really know how the distributions of their features are, then they can try to use these optimizations.

Reply:
> but we don't, cause in many cases those outliers are important somehow

But even if they are important or not, I don't see why we would not recommend the `RobustScaler`. To me, a `QuantileTransformer` would be the one completely neglecting that an outlier is an outlier. The fact that `RobustScaler` is a linear transformation would not have this effect.

Reply:
I don't mind recommending RobustScaler, but if we want to include in the example the analysis here, that we actually checked and the solution to the problem was the same with GridSearchCV or not, we should not use something which might leak data. Even then, why would we _not_ stay on the safer side? People write code, then deploy the code, and it stays there and re-trains the model a year after they last looked at the actual data, not knowing if the assumptions they made to skip the GridSearchCV still hold.

Reply:
Now, I get the comment of @ogrisel :). So let's do an outer grid-search.